### ISSUE TITLE
security group rule creation failing - endpoint has been discontinued on 2022-03-29 

### ISSUE THREAD

**Issue by powellquiring** [IBM/packer-plugin-ibmcloud#43]
**Describe the bug**
```
2022/03/29 13:12:25 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 2022/03/29 13:12:25 Response Body from ibmcloud-  {"port_max":22,"port_min":22,"protocol":"tcp","direction":"inbound","href":"https://us-south.iaas.cloud.ibm.com/v1/security_groups/r006-dec0417e-ce05-4f39-bcb9-e2fb8e6c00be/rules/r006-3333f316-eb6a-4f52-9934-6ff95b93ad91","id":"r006-3333f316-eb6a-4f52-9934-6ff95b93ad91","ip_version":"ipv4","protocol":"tcp","remote":{"cidr_block":"0.0.0.0/0"}}
2022/03/29 13:12:25 ui: [1;32m==> ibmcloud-vpc.instance: ssh rule ID: r006-3333f316-eb6a-4f52-9934-6ff95b93ad91[0m
2022/03/29 13:12:25 ui: [1;32m==> ibmcloud-vpc.instance: Attaching Instance to the Security Group[0m
2022/03/29 13:12:25 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 2022/03/29 13:12:25 Response Status -  404
2022/03/29 13:12:25 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 2022/03/29 13:12:25 Response Body from ibmcloud-  {"errors":[{"code":"not_found","message":"This endpoint has been discontinued on 2022-03-29 00:00:00 +0000 UTC"}],"trace":"730fc7e7-f407-48a3-af50-586c6839207c"}
2022/03/29 13:12:25 ui: [1;32m==> ibmcloud-vpc.instance: Releasing the Floating IP: 52.118.184.253 ...[0m
2022/03/29 13:12:25 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 2022/03/29 13:12:25 Response Status -  200

2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: panic: interface conversion: interface {} is nil, not string
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: goroutine 7 [running]:
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: packer-plugin-ibmcloud/builder/ibmcloud/vpc.(*stepCreateSecurityGroupRules).Run(0x18996c8, 0x12315e8, 0xc000510080, 0x1231a48, 0xc0002f41b0, 0x0)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	packer-plugin-ibmcloud/builder/ibmcloud/vpc/step_create_sec_group_rules.go:105 +0x1397
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc0002f4240, 0x12315e8, 0xc000510080, 0x1231a48, 0xc0002f41b0)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.2.3/multistep/basic_runner.go:70 +0x234
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: packer-plugin-ibmcloud/builder/ibmcloud/vpc.(*Builder).Run(0xc000514a00, 0x12315e8, 0xc000510080, 0x12362e8, 0xc0002f4150, 0x121c800, 0xc00011c020, 0xc0002f4000, 0xf083c0, 0xc00011c000, ...)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	packer-plugin-ibmcloud/builder/ibmcloud/vpc/builder.go:104 +0x3e2
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc00030d8c0, 0x1, 0xc0002ea010, 0x0, 0x0)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.2.3/rpc/builder.go:117 +0x1c4
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: reflect.Value.call(0xc000301500, 0xc0003084c8, 0x13, 0x1038848, 0x4, 0xc0002f2f08, 0x3, 0x3, 0xc000048000, 0xc0004856f8, ...)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	reflect/value.go:476 +0x8e7
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: reflect.Value.Call(0xc000301500, 0xc0003084c8, 0x13, 0xc000485708, 0x3, 0x3, 0xc000580d38, 0xc000584410, 0x461f20)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	reflect/value.go:337 +0xb9
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00030d900, 0xc000318c30, 0xc00030b548, 0xc00030b550, 0xc000310e80, 0xc00030f340, 0xea8960, 0xc0002ea00c, 0x18a, 0xe8db40, ...)
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:377 +0x189
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec
2022/03/29 13:54:02 packer-plugin-ibmcloud_v2.1.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:474 +0x44d

```

**To Reproduce**
Any pkr.hcl file is likely to fail.  Here is mine:

```
packer {
  required_plugins {
    ibmcloud = {
      version = ">=v2.1.0"
      source  = "github.com/IBM/ibmcloud"
    }
  }
}

variable "ibmcloud_api_key" {
  type = string
}

variable "region" {
  type = string
}

variable "subnet_id" {}
variable "resource_group_id" {}
variable "security_group_id" { default = "" }

variable "base_image_name" { default = "ibm-rocky-linux-8-5-minimal-amd64-2" }
variable "profile" { default = "bx2-2x8" }
variable "image_name" {
  default = ""
}

locals {
  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
  image_name = var.image_name == "" ? "packer-${local.timestamp}" : var.image_name
}

source "ibmcloud-vpc" "instance" {
  api_key = var.ibmcloud_api_key
  region  = var.region

  subnet_id         = var.subnet_id
  resource_group_id = var.resource_group_id
  security_group_id = var.security_group_id

  vsi_base_image_name = var.base_image_name
  vsi_profile         = var.profile
  vsi_interface       = "public"
  vsi_user_data_file  = ""

  image_name = local.image_name

  communicator = "ssh"
  ssh_username = "root"
  ssh_port     = 22
  ssh_timeout  = "15m"

  timeout = "30m"
}

build {
  sources = [
    "source.ibmcloud-vpc.instance"
  ]

  provisioner "shell" {
    inline = [ "mkdir -p /usr/local/on-boot/" ]
  }

}

```

**Expected behavior**

**Screenshots**

**Comment by syyang-in-cloud** [comment]
Thanks @powellquiring for filing this bug; I'm experiencing the same issue.

**Comment by juanpinzon** [comment]
The problem is due an API update on IBM Cloud VPC. Currently the versioning is generated based on the current date. however, according with the documentation (https://cloud.ibm.com/apidocs/vpc#api-versioning), we should be hard-coding the version in our code and update and test the plugin with new version dates to maintain ongoing compatibility.

**Comment by vicyou** [comment]
i am experiencing the same issue. when can this be fixed? thanks.

**Comment by juanpinzon** [comment]
There are multiple changes on the new API, I am aware of such changes and will work to fix the issues next week.

**Comment by juanpinzon** [comment]
Problems solved in the new plugin release v2.2.0
